security/s2a/internal/crypter/counter.go

```diff
@@ -0,0 +1,42 @@
+package crypter
+
+import "errors"
+
+const counterLen = 12
+
+// counter is a 96-bit, little-endian counter.
+type counter struct {
+	value   [counterLen]byte
+	invalid bool
+}
+
+func newCounter() counter {
+	return counter{}
+}
+
+// val returns the current value of the counter as a byte slice.
+func (c *counter) val() ([]byte, error) {
+	if c.invalid {
+		return nil, errors.New("invalid counter, possibly due to overflow")
+	}
+	return c.value[:], nil
+}
+
```

Comment:
Please add a `reset()` API that sets the sequence number to zero.

Reply:
Done